Actualize README files for ubuntu (#93)
GreengageDB/greengage: README.md

### Installing dependencies (for Linux developers)
Follow [appropriate linux steps](README.linux.md) for getting your system ready for GPDB

### Downloading submodules
Make sure that you download submodules. To do this, use `git clone --recurse-submodules` 
when downloading the source code. If you want to update the submodules, run:
```bash
git submodule update --init --recursive --force
```

### Build the database

> The recommended way to build the database is to use build system located in gpAux 

Reviewer: ```suggestion
The recommended way to build the database is to use build system located in the gpAux 
```

directory, which is also used for CI testing and building packages.

To make optimized release build use the following:

```
make GPROOT=~/build PARALLEL_MAKE_OPTS=-j8 dist -C gpAux
```

In order to run regression tests debug build is required with debug extensions. It can be built using the following command:

```
make GPROOT=~/build PARALLEL_MAKE_OPTS=-j8 devel -C gpAux
```

Bring in greengage environment into your running shell:

```

> source $HOME/build/greengage-db-dist/greengage_path.sh

Reviewer 2: I think better to save both variant:
- greengage-db-dist    -> for `make dist`
- greengage-db-devel -> for `make devel`

```

Start demo cluster:

```
make create-demo-cluster
```

> To use demo cluster use environment variables in gpdemo-env.sh which contains 

Reviewer: ```suggestion
To use the demo cluster, source the environment variables from gpdemo-env.sh, which contains 
```

__PGPORT__ and __MASTER_DATA_DIRECTORY__ values:

```
source gpAux/gpdemo/gpdemo-env.sh
```

The directory, the TCP ports, the number of segments, and the existence of
standbys for segments and coordinator for the demo cluster can be changed
when starting demo cluster. 
Instead of `make create-demo-cluster`, consider:

```
DATADIRS=/tmp/gpdb-cluster PORT_BASE=5555 NUM_PRIMARY_MIRROR_PAIRS=1 WITH_MIRRORS=false make create-demo-cluster
```

If you want to clean all generated files:
```
make distclean
```

## Running tests

> * By default tests are using GPORCA optimizer:

Reviewer: ```suggestion
* By default, tests use the GPORCA optimizer:
```

# Configure build environment to install at /usr/local/gpdb
./configure --with-perl --with-python --with-libxml --with-gssapi --prefix=/usr/local/gpdb

# Note that to run the full test suite (see below), fault injector is required, 
and the following configure command line is required:

./configure \
    --enable-debug --enable-orca --enable-debug-extensions \
    --with-openssl --with-perl --with-python \
    --enable-gpfdist \
    --with-libxml --with-gssapi --prefix=/usr/local/gpdb

# Compile and install
make -j8

> make install

Reviewer 3: Why are theese changes needed? There are compiller dependent steps during install. At least for python modules.

Author: This fragment totally rewritten with using of gpAux/Makefile

DATADIRS=/tmp/gpdb-cluster PORT_BASE=5555 NUM_PRIMARY_MIRROR_PAIRS=1 WITH_MIRRORS=false make create-demo-cluster
```

If you want to clean all generated files
```
make distclean
```

## Running tests

* Tests with GPORCA are not using table alias, so one can run them with table alias off:
```

> PGOPTIONS='-c optimizer_enable_table_alias=false' make installcheck-world

Reviewer 3: I suggest moving this GUC to the other test environment specific params for demo stand (see CLUSTER_CONFIG_POSTGRES_ADDONS).

To turn GPORCA off and use Postgres planner for query optimization:
```

> PGOPTIONS='-c optimizer=off' make installcheck-world

Author: added Orca run

## Alternative Configurations

Internally, the configure script is used to adapt the build system to the characteristics of the host machine. For finer control over the components that are built, configuration options must be specified explicitly. The configuration options in effect are recorded in the file config.log and may be inspected with the following command:
```
head config.log
```

### Building GPDB without GPORCA

Currently, GPDB is built with GPORCA by default. If you want to build GPDB

> without GPORCA, configure requires `--disable-orca` flag to be set.

Reviewer 3: > results in having both --enable-orca and --disable-orca in configure command line

Yes, but this combination is handled properly and lead to build without Orca. But it's up to you.
